### ISSUE TITLE
Tensorflow 将ckpt 模型 转化成.pb文件方便使用

### ISSUE THREAD

**Issue by beijinggao** [beijinggao/laogaoganhuo#2]
import tensorflow as tf
a = tf.constant(1.3, name='const_A')
b = tf.Variable(3.1, name='b')
c = tf.add(a, b, name='addition')
d = tf.multiply(c, a, name='multiply')

for op in tf.get_default_graph().get_operations():
    print str(op.name)

**Comment by beijinggao** [comment]
#encoding=utf8
 import tensorflow as tf
 import numpy as np
from inception import inception_model as inception
 import os
 import image_processing
 from tensorflow.python.framework  import graph_util
 from tensorflow.python.platform import gfile

 checkpoint_dir="/data1/gao/tensorflow/slim/models/inception/carmodel1/model.ckpt-100000"
 imagePath="/data1/gao/tensorflow/slim/models/inception/16.jpg"
 image_data = tf.gfile.FastGFile(imagePath, 'rb').read()
 def predict():
   with tf.Graph().as_default():
      images=image_processing.decode_jpeg(image_data)
      images=image_processing.eval_image(images,299,299)
      images = tf.expand_dims(images,0)
      images_re = tf.reshape(images, shape=[1, 299, 299, 3])
      logits, _ = inception.inference(images_re,34)
      print('#####################################')
      print(logits)
      variable_averages = tf.train.ExponentialMovingAverage(inception.MOVING_AVERAGE_DECAY)
      variables_to_restore = variable_averages.variables_to_restore()
      saver = tf.train.Saver(variables_to_restore)
      #saver = tf.train.Saver(sharded=True)
      with tf.Session() as sess:
        saver.restore(sess, checkpoint_dir)
        predictions=sess.run([logits])
        predictions = np.squeeze(predictions)
        top_k = predictions.argsort()[-5:][::-1]  # Getting top 5 predictions
        for node_id in top_k:
          score = predictions[node_id]
          print('%s (score = %.5f)' % (node_id,score))
        graph_def = tf.get_default_graph().as_graph_def()
        save_graph(graph_def,sess)
 
 
  def save_graph(graph_def,sess):
      # Write out the trained graph and labels with the weights stored as constants.
      output_graph_def = graph_util.convert_variables_to_constants(
        sess, graph_def,['inception_v3/logits/final_result_1'])
      with gfile.FastGFile("./aaaatest.pb", 'wb') as f:
          f.write(output_graph_def.SerializeToString())
 
 
  if __name__=="__main__":
    predict()




  1 #encoding=utf8
  2 import numpy as np
  3 import tensorflow as tf
  4 import shutil
  5 import os
  6 import datetime
  7
  8 imagePath = './15.jpg'
  9 modelFullPath = './aaaatest.pb'
 10 labelsFullPath = './tmp_model/output_labels.txt'
 11
 12
 13 def create_graph():
 14     """Creates a graph from saved GraphDef file and returns a saver."""
 15     # Creates graph from saved graph_def.pb.
 16     with tf.gfile.FastGFile(modelFullPath, 'rb') as f:
 17         graph_def = tf.GraphDef()
 18         graph_def.ParseFromString(f.read())
 19         _ = tf.import_graph_def(graph_def, name='')
 20
 21
 22 def run_inference_on_image():
 23     image_data = tf.gfile.FastGFile(imagePath, 'rb').read()
 24     with tf.Session() as sess:
 25         try:
 26             softmax_tensor = sess.graph.get_tensor_by_name('inception_v3/logits/final_result_1:0')
 27             predictions = sess.run(softmax_tensor,
 28                                {'decode_jpeg/DecodeJpeg/contents:0': image_data})
 29             predictions = np.squeeze(predictions)
 30             top_k = predictions.argsort()[-5:][::-1]  # Getting top 5 predictions
 31             for node_id in top_k:
 32                 score = predictions[node_id]
 33                 print('%s (score = %.5f)' % (node_id,score))
 34         except Exception, e:
 35             print e.message
 36             print("error!!")
 37
 38 def ensure_dir_exists(dir_path):
 39     isExist_y=os.path.exists(dir_path)
 40     if (not isExist_y):
 41         os.makedirs(dir_path)
 42
 43 if __name__ == '__main__':
 44     # Creates graph from saved GraphDef.
 45     create_graph()
 46     run_inference_on_image()
